Bottom: bb5e86f4480af409b6c351dbd8bc0203c9fcb215
Top:    7a03fb735d76e7a41eeb96fe6a2d61e8de586a43
Author: Lawrence D'Anna <lawrence_danna@apple.com>
Date:   2019-10-17 12:09:28 -0700

Refresh of eliminate-one-form-of

---
diff --git a/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.h b/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.h
index d34b96c566f..0cdb63f17c9 100644
--- a/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.h
+++ b/lldb/source/Plugins/ScriptInterpreter/Python/PythonDataObjects.h
@@ -232,15 +232,9 @@ public:
     return result;
   }
 
-  PythonObject &operator=(const PythonObject &other) {
-    Reset(PyRefType::Borrowed, other.get());
-    return *this;
-  }
-
-  PythonObject &operator=(PythonObject &&other) {
+  PythonObject &operator=(PythonObject other) {
     Reset();
-    m_py_obj = other.m_py_obj;
-    other.m_py_obj = nullptr;
+    m_py_obj = std::exchange(other.m_py_obj, nullptr);
     return *this;
   }
